### PR TITLE
found a scenario where the data table search box can't search the val…

### PR DESCRIPTION
**Pull request by Sujon94**
…ue consisting of '&' though the data exists in the table. Fixed the issue.

### CODE REVIEW

File: src/DataTableAbstract.php [hunk 1]
```diff
@@ -737,6 +737,8 @@ protected function filterRecords(): void
     public function filtering(): void
     {
         $keyword = $this->request->keyword();
+        /**Following line of code to convert & or any other special character to html string. **/
+        $keyword = (is_string($keyword) || $keyword instanceof Htmlable) ? e($keyword) : $keyword;
```
Comment (yajra) [inline src/DataTableAbstract.php:741]: `$keyword` is always a string thus Htmlable is not needed. Maybe we can just convert to HTML string directly from the `request->keyword()` method?

**Review by yajra — CHANGES_REQUESTED** [review]

File: src/DataTableAbstract.php (hunk 1 above)
Comment (Sujon94) [reply src/DataTableAbstract.php:741]: Yes, you are right. We cam ommit the conditional part and directly convert to HTML string.

Comment (Sujon94) [reply src/DataTableAbstract.php]: I have commited. Please take a review.

**Comment by yajra** [comment]
The changes look good. However, this only changes the filtering keyword. Doing a column search would yield the previous results. 

Would you be able to review this PR https://github.com/yajra/laravel-datatables/pull/2903? That PR will make the keyword to HTML entities anywhere it is used.

**Comment by Sujon94** [comment]
Ok, I will review.

**Comment by yajra** [comment]
Closing in favor of https://github.com/yajra/laravel-datatables/pull/2903.

**Pull request closed without merging**